sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -605,7 +605,7 @@
         },
         "defaultPEConnections": "[array(variables('privateEndpointSettings'))]",
         "privateEndpointDeploymentName": "[concat('DeployPrivateEndpoint-', uniqueString(parameters('privateEndpointName')))]",
-        "materialization_identity_resource_id": "[if(not(equals(parameters('materialization_identity_name'), 'empty')), resourceId(resourceGroup().name, 'Microsoft.ManagedIdentity/userAssignedIdentities', parameters('materialization_identity_name')),  parameters('materialization_identity_resource_id'))]",
+        "materialization_identity_resource_id": "[if(and(empty(parameters('materialization_identity_resource_id')), equals(parameters('set_up_feature_store'), 'true')), resourceId(resourceGroup().name, 'Microsoft.ManagedIdentity/userAssignedIdentities', parameters('materialization_identity_name')),  parameters('materialization_identity_resource_id'))]",
```

Comment:
Is this used in the condition at line 657 below?

Reply:
yes, we have parameters and variables

Reply:
If this is the case, then line 657 condition will never be true because empty(parameters('materialization_identity_resource_id')) will always be false. Is it expected?